src/components/react-json-schemaForm/TableField/Tbody/Tbody.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from '@blueprintjs/core';
+import uuid from 'uuid/v4';
+
+import { getObjectOrderedValue } from '../../../../modules/CRUD/services/utils';
+import Field from  '../Field';
+
+const Tbody = ({
+  disabled,
+  formData,
+  onChange,
+  registry,
+  schema: {
+    items: {
+      properties,
+      required,
+    },
+  },
+  uiSchema: {
+    items: uiSchemaItems,
+  },
+  onUpdatingRow,
+}) => {
+  const [data, setData] = useState(formData);
+
+  useEffect(() => {
+    onUpdatingRow(data);
+  }, [data, onUpdatingRow]);
+
+
+  const handleUpdatingRow = useCallback((index, action) => {
+    setData(prevData => {
+      const cloneData = [...prevData];
+      if (action === 'inserting') {
+        const orderedValue = getObjectOrderedValue(properties, uiSchemaItems['ui:order']);
+        const emptyRow = Object.keys(orderedValue).reduce((list, key) => ({
+          ...list, [key]: undefined,
+        }), {});
+        cloneData.splice(index, 0, emptyRow);
+      }
+      if (action === 'removing') {
+        cloneData.splice(index, 1);
+      }
+      if (action === 'moving-up') {
+        cloneData.splice(index - 1, 0, cloneData.splice(index, 1)[0]);
+      }
+      if (action === 'moving-down') {
+        cloneData.splice(index + 1, 0, cloneData.splice(index, 1)[0]);
+      }
+      return cloneData;
+    });
+  }, [properties, uiSchemaItems]);
+
+
+  if (!data.length) {
+    handleUpdatingRow(0, 'inserting');
+  }
+
+  return (
+    <tbody>
+      {data.map((item, index) => {
+        const orderedValues = getObjectOrderedValue(properties, uiSchemaItems['ui:order']);
```

Comment:
Why is this computed for *each* iteration of `data.map` when it does not use any loop data?

Reply:
Right, this is a mistake.
Moreover, it's also declared line 36.

I'll refacto and memoize the constante at the top of the component 👍